test/integration/compliance_history_test.go

```diff
@@ -777,8 +777,42 @@ var _ = Describe("GRC: [P1][Sev1][policy-grc] Test the compliance history API",
 		 }]`)
 		Expect(err).ToNot(HaveOccurred())
 
-		By("The Policy should be Compliant")
-		clusters = verifyPolicyOnAllClusters(ctx, policyNS, policyName, "Compliant", defaultTimeoutSeconds*2)
+		debugMsg := common.RegisterDebugMessage()
+
+		By("The Policy should be Compliant on all clusters")
+		Eventually(func(g Gomega) {
+			clusters = confirmComplianceOnAllClusters(ctx, policyNS, policyName, "Compliant")(g)
+
+			*debugMsg = ""
+
+			for _, cluster := range clusters {
+				replPol, err := clientHubDynamic.Resource(common.GvrPolicy).Namespace(cluster).Get(
+					ctx, policyNS+"."+policyName, metav1.GetOptions{},
+				)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(replPol).NotTo(BeNil())
+
+				details, found, err := unstructured.NestedSlice(replPol.Object, "status", "details")
+				g.Expect(found).To(BeTrue())
+				g.Expect(err).NotTo(HaveOccurred())
+
+				det, ok := details[0].(map[string]interface{})
+				g.Expect(ok).To(BeTrue())
+
+				history, found, err := unstructured.NestedSlice(det, "history")
+				g.Expect(found).To(BeTrue())
+				g.Expect(err).NotTo(HaveOccurred())
+
+				item, ok := history[0].(map[string]interface{})
+				g.Expect(ok).To(BeTrue())
+
+				msg, found, err := unstructured.NestedString(item, "message")
+				g.Expect(found).To(BeTrue())
+				g.Expect(err).NotTo(HaveOccurred())
+
+				*debugMsg += fmt.Sprintf("Cluster %v current compliance message: %v\n", cluster, msg)
```

Comment:
Will this code ever be reached if there is a failure in `confirmComplianceOnAllClusters` since a `g.Expect` would have failed?

Reply:
Ah, you're right, I should've put them in the other order.